SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -811,12 +811,15 @@ Plotter {
 
 + Function {
 	plot { |duration = 0.01, target, bounds, minval, maxval, separately = false|
-		var name = this.asCompileString, plotter;
+
+		var name = this.asCompileString, plotter, selector;
 		if(name.size > 50 or: { name.includes(Char.nl) }) { name = "function plot" };
 		plotter = Plotter(name, bounds);
 		plotter.value = [0.0];
+		target = target.asTarget;
+		selector = if(target.server.isLocal) { \loadToFloatArray } { \getToFloatArray };
 
-		this.getToFloatArray(duration, target, { |array, buf|
+		this.perform(selector, duration, target, action: { |array, buf|
```

Comment:
I feel this would be much easier to understand (and more readable in general) if it used actual function calls rather than `perform`. same below

Reply:
Factoring out in this case really impairs readability, though. This is exactly what I had in mind.

Reply:
Maybe it would be clear if I renamed "selector"?

`this.perform(dataTransmissionMethod, duration, target, ` ?

Reply:
No, it's the control flow and not the variable names that is confusing